pkg/azure/blob/credentials_test.go

```diff
@@ -0,0 +1,132 @@
+package blob
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"get.porter.sh/plugin/azure/pkg/azure/azureconfig"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGet_GetCredentials(t *testing.T) {
+	testcases := []struct {
+		name         string
+		envVarsToSet map[string]string
+		config       *azureconfig.Config
+		testfunc     func(t *testing.T, envVarsToSet map[string]string, config *azureconfig.Config, logger hclog.Logger)
+	}{
+		{
+			"Missing Environment Variables",
+			map[string]string{},
+			&azureconfig.Config{},
+			missingEnvironmentVariables,
+		},
+		{
+			"Invalid Connection string",
+			map[string]string{
+				"AZURE_STORAGE_CONNECTION_STRING": "Invalid",
+			},
+			&azureconfig.Config{},
+			invalidConnectionString,
+		},
+		{
+			"Valid Connection string",
+			map[string]string{
+				"AZURE_STORAGE_CONNECTION_STRING": "DefaultEndpointsProtocol=https;AccountName=bmFtZQo=;AccountKey=a2V5Cg==;EndpointSuffix=core.windows.net",
+			},
+			&azureconfig.Config{},
+			validConnectionString,
+		},
+		{
+			"Missing Storage Acccount Resource Group",
+			map[string]string{},
+			&azureconfig.Config{
+				StorageAccount: "account",
+			},
+			missingStorageAccountResourceGroup,
+		},
+		{
+			"Missing Storage Acccount Name",
+			map[string]string{},
+			&azureconfig.Config{
+				StorageAccountResourceGroup: "group",
+			},
+			missingStorageAccountName,
+		},
+		{
+			"loginwithCLI",
+			map[string]string{},
+			&azureconfig.Config{
+				StorageAccount:              "account",
+				StorageAccountResourceGroup: "group",
+			},
+			loginwithCLI,
+		},
+	}
+	env := os.Environ()
+	for _, tc := range testcases {
+
+		t.Run(tc.name, func(t *testing.T) {
+
+			logger := hclog.New(&hclog.LoggerOptions{
+				Name:   strings.ReplaceAll(tc.name, " ", "_"),
+				Output: os.Stderr,
+				Level:  hclog.Error,
+			})
+
+			for k, v := range tc.envVarsToSet {
+				os.Setenv(k, v)
+			}
+
+			tc.testfunc(t, tc.envVarsToSet, tc.config, logger)
+			resetEnvironmentVars(t, env)
```

Comment:
This is problemmatic for a couple reasons:

* If testFunc does not execute successfully and the test exits at this line, the env vars are not unset
* os.ClearEnv unsets ALL env vars, not just the ones set by the test

A more preferred pattern to deal with setting/unsetting env vars in a test would be 

```go
for k, v := range tc.envVarsToSet {
    os.Setenv(k, v)
}
defer func() {
  for k := range tc.envVarsToSet {
    os.Unset(k)
}
}()
```

```

Reply:
I updated it to use Unsetenv,  and made sure that it executed when a test failed with defer, the os.Clearenv was not a problem as env contained the environment before the tests ran